tests/unit/helpers/contracts.test.js

```diff
@@ -538,6 +542,24 @@ describe('endContract', () => {
         { query: 'lean', args: [{ autopopulate: true, virtuals: true }] },
       ]
     );
+    SinonMongoose.calledWithExactly(
```

Comment:
Il te faut 2 tests en plus pour vérifier :
- que tu as bien une 403 si tu as des `eventsHistories`
- que tu as bien une 409 si ta `endDate` est avant la `lastVersion.startDate`

Reply:
Tu as un très bon exemple de test unitaire avec des try / catch dans sectorHistories.test.js l.361

Reply:
Oui bien vu j'ai ajouté le test pour la 403. Par contre le second test (pour la 409) est deja existant: `should throw an error if contract end date is before last version start date`